### ISSUE TITLE
Update spec.rule.http to be optional for Nginx's master mergable ingress

### ISSUE THREAD

**Issue by riesbl01** [hashicorp/terraform-provider-kubernetes#1199]
### Terraform Version, Provider Version and Kubernetes Version

```
Terraform version: Terraform v0.13.5
Kubernetes provider version: provider registry.terraform.io/hashicorp/kubernetes v2.0.2
Kubernetes version: 1.17.16
```

### Affected Resource(s)

- kubernetes_ingress

### Terraform Configuration Files
```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```
resource "kubernetes_ingress" "nginx-master-ingress" {
  depends_on = [helm_release.nginx-lb]

  wait_for_load_balancer = true
  metadata {
    name = "nginx-ingress-master"
    namespace = "nginx-ingress"
    annotations = {
      "kubernetes.io/ingress.class" = "nginx"
      "nginx.org/mergeable-ingress-type" = "master"
    }
  }
  spec {
    rule {
      host = <host> 
    }
  }
}

### Panic Output

N/A

### Expected Behavior
What should have happened?
Expected http attribute to be optional so that a ingress can be deployed with just a host which is a requirement of the master mergeable ingress. 

### Actual Behavior
What actually happened?
```
Error: Required attribute is not set

  on main.tf line 222, in resource "kubernetes_ingress" "nginx-master-ingress":
 222:     rule {
```
### Important Factoids

N/A

### References

- https://github.com/nginxinc/kubernetes-ingress/tree/v1.10.0/examples/mergeable-ingress-types 
- https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/ingress#http

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by riesbl01** [comment]
Also tried to setup a rule for the nginx controller but the ingress fails because the master ingress file can only have a hostname. A path and backend are now allowed. ERROR:
"Warning  Rejected  4m32s  nginx-ingress-controller  spec.rules[0].http.paths: Too many: 1: must have at most 0 items"

**Comment by skleinsmith** [comment]
Upvoted as I have the same challenge. I'm using the AWS Load Balancer controller and want to create an empty ingress so that the load balancer can get created (as it won't use one that it didn't create). The below snippet works using kubectl, but can't be re-created in Terraform due to `spec.rule.http` being required.

``` yaml
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: ingress-test-hello-world
  labels:
    helm.sh/chart: hello-world-0.1.0
    app.kubernetes.io/name: hello-world
    app.kubernetes.io/version: "1.0.0"
    app.kubernetes.io/managed-by: Helm
  annotations:
    kubernetes.io/ingress.class: alb
    alb.ingress.kubernetes.io/scheme: internal
    alb.ingress.kubernetes.io/target-type: ip
    alb.ingress.kubernetes.io/backend-protocol: HTTP
    alb.ingress.kubernetes.io/group.name: test-ingress
spec:
  rules:
    - host:
```

**Comment by Zakay** [comment]
👍

**Comment by arybolovlev** [comment]
Hi folks,

This issue has been addressed in `v2.9.0`. I will go ahead and close this issue.

Thanks, everyone for your patience.
